code/game/machinery/doors/firedoor.dm

```diff
@@ -383,6 +385,13 @@
 				for(var/i=1;i<=ALERT_STATES.len;i++)
 					if(dir_alerts[d] & (1<<(i-1)))
 						overlays += new/icon(icon,"alert_[ALERT_STATES[i]]", dir=cdir)
+		if (hashatch)
+			hatch_image.color = hatch_colour//This line is unnecessary, but is here for configuring colours ingame. Should be removed when finished
+			if (hatchstate)
+				hatch_image.icon_state = "[hatchstyle]_open"
+			else
+				hatch_image.icon_state = hatchstyle
+			overlays += hatch_image

```

Comment:
Doesn't this keep expanding the overlays list ad infinitum? You're adding the same reference to it over and over again.

Either simply add a, `overlays -= hatch_image` to remove the old reference first, or identify if a chance is necessary. Both are fine, but this specific form seems bad.